guava/src/com/google/common/io/BaseEncoding.java

```diff
@@ -870,10 +887,13 @@ public int hashCode() {
   static final class Base16Encoding extends StandardBaseEncoding {
     final char[] encoding = new char[512];
 
-    Base16Encoding(String name, String alphabetChars) {
+    Base16Encoding(String name, @MinLen(1) String alphabetChars) {
       this(new Alphabet(name, alphabetChars.toCharArray()));
     }
 
+    @SuppressWarnings({"array.access.unsafe", "argument.type.incompatible"}) /* Since encoding has a length of 512, it is safe to
```

Comment:
Is `encoding` annotated as `@ArrayLen(512)` or `@MinLen(512)`? If so I would think this ought to be verifiable, though the Index Checker might not be sufficiently precise with bitwise ops.

Reply:
Annotating `encoding` with `@ArrayLen(512)` doesn't fix the error. I agree that the checker is imprecise in bitwise operations.